x/exchange/msg_test.go

```diff
@@ -2491,6 +2510,637 @@
+				SourceAmount: ValidPayment.SourceAmount,
+				Target:       ValidPayment.Target,
+				TargetAmount: ValidPayment.TargetAmount,
+				ExternalId:   "",
+			}},
+			expErr: nil,
+		},
+		{
+			name: "invalid payment",
+			msg: MsgCreatePaymentRequest{Payment: Payment{
+				Source:       "",
+				SourceAmount: ValidPayment.SourceAmount,
+				Target:       "notgoodeither",
+				TargetAmount: ValidPayment.TargetAmount,
+				ExternalId:   ValidPayment.ExternalId,
+			}},
+			expErr: []string{
+				"invalid source \"\": empty address string is not allowed",
+				"invalid target \"notgoodeither\": decoding bech32 failed: invalid separator index -1",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			testValidateBasic(t, &tc.msg, tc.expErr)
+		})
+	}
+}
+
+func TestMsgAcceptPaymentRequest_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name   string
+		msg    MsgAcceptPaymentRequest
+		expErr []string
+	}{
+		{
+			name:   "valid payment",
+			msg:    MsgAcceptPaymentRequest{Payment: ValidPayment},
+			expErr: nil,
+		},
+		{
+			name: "invalid payment",
+			msg: MsgAcceptPaymentRequest{Payment: Payment{
+				Source:       "",
+				SourceAmount: ValidPayment.SourceAmount,
+				Target:       "notgoodeither",
+				TargetAmount: ValidPayment.TargetAmount,
+				ExternalId:   ValidPayment.ExternalId,
+			}},
+			expErr: []string{
+				"invalid source \"\": empty address string is not allowed",
+				"invalid target \"notgoodeither\": decoding bech32 failed: invalid separator index -1",
+			},
+		},
+		{
+			name: "no target",
+			msg: MsgAcceptPaymentRequest{Payment: Payment{
+				Source:       ValidPayment.Source,
+				SourceAmount: ValidPayment.SourceAmount,
+				Target:       "",
+				TargetAmount: ValidPayment.TargetAmount,
+				ExternalId:   ValidPayment.ExternalId,
+			}},
+			expErr: []string{"invalid target: empty address string is not allowed"},
+		},
+		{
+			name: "no source amount",
+			msg: MsgAcceptPaymentRequest{Payment: Payment{
+				Source:       ValidPayment.Source,
+				SourceAmount: nil,
+				Target:       ValidPayment.Target,
+				TargetAmount: ValidPayment.TargetAmount,
+				ExternalId:   ValidPayment.ExternalId,
+			}},
+			expErr: nil,
+		},
+		{
+			name: "no target amount",
+			msg: MsgAcceptPaymentRequest{Payment: Payment{
+				Source:       ValidPayment.Source,
+				SourceAmount: ValidPayment.SourceAmount,
+				Target:       ValidPayment.Target,
+				TargetAmount: nil,
+				ExternalId:   ValidPayment.ExternalId,
+			}},
+			expErr: nil,
+		},
+		{
+			name: "no external id",
+			msg: MsgAcceptPaymentRequest{Payment: Payment{
+				Source:       ValidPayment.Source,
+				SourceAmount: ValidPayment.SourceAmount,
+				Target:       ValidPayment.Target,
+				TargetAmount: ValidPayment.TargetAmount,
+				ExternalId:   "",
+			}},
+			expErr: nil,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			testValidateBasic(t, &tc.msg, tc.expErr)
+		})
+	}
+}
```

Comment:
Payments are allowed to have an empty `target`, or `external_id`. They are also allowed to not have a  `source_amount` or `target_amount`. In a `MsgAcceptPaymentRequest`, though, a `target` **is** required. But it would be incorrect to update `ValidateBasic` to also require those other fields.